client/signup/jetpack-connect/plans.jsx

```diff
@@ -60,6 +60,39 @@ const Plans = React.createClass( {
 			( site.plan.product_slug === 'jetpack_business' || site.plan.product_slug === 'jetpack_premium' );
 	},
 
+	selectFreeJetpackPlan() {
+		const selectedSite = this.props.sites.getSelectedSite();
+		this.props.recordTracksEvent( 'calypso_jpc_plans_submit_free', {
+			user: this.props.userId
+		} );
+		if ( isCalypsoStartedConnection( this.props.jetpackConnectSessions, selectedSite.slug ) ) {
+			page.redirect( CALYPSO_REDIRECTION_PAGE + selectedSite.slug );
+		} else {
+			const { queryObject } = this.props.jetpackConnectAuthorize;
+			this.props.goBackToWpAdmin( queryObject.redirect_after_auth );
+		}
+	},
+
+	selectPlan( cartItem ) {
+		const selectedSite = this.props.sites.getSelectedSite();
+		const checkoutPath = `/checkout/${ selectedSite.slug }`;
+		if ( cartItem.product_slug === 'jetpack_free' ) {
+			return this.selectFreeJetpackPlan();
+		}
+		if ( cartItem.product_slug === 'jetpack_premium' ) {
+			this.props.recordTracksEvent( 'calypso_jpc_plans_submit_99', {
+				user: this.props.userId
+			} );
+		}
+		if ( cartItem.product_slug === 'jetpack_business' ) {
+			this.props.recordTracksEvent( 'calypso_jpc_plans_submit_299', {
+				user: this.props.userId
+			} );
+		}
+		upgradesActions.addItem( cartItem );
```

Comment:
I tried that, but in this context we do not have access to `plan`, so we cannot `getPlanPath( plan )`
Is this a problem? Are we getting rid of `upgradeActions`